src/shared/utils/metrics.ts

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable camelcase */
+import { metrics as sentryMetrics } from '@sentry/react'
+
+type LeafKeyUnion<T, K extends string = ''> = {
```

Comment:
Can we call this `MetricKeyNameUnion` or something similar so it's more descriptive?

Reply:
Ehhhhhhhhh yea I guess that's okay